Makefile

```diff
@@ -20,11 +19,14 @@ $(UTILS)/cpuinfo.o : $(UTILS)/cpuinfo.cpp
 $(TESTDIR)/%.o : $(TESTDIR)/%.cpp $(SRCS)
 		$(CXX) $(CXXFLAGS) $(MARCHFLAG) $(GTESTCFLAGS) -c $< -o $@
 
-test: $(TESTDIR)/main.cpp $(TESTOBJS) $(UTILS)/cpuinfo.o $(SRCS)
-		$(CXX) tests/main.cpp $(TESTOBJS) $(UTILS)/cpuinfo.o $(MARCHFLAG) $(CXXFLAGS) $(GTESTLDFLAGS) -o testexe
+test: $(TESTOBJS) $(UTILS)/cpuinfo.o $(SRCS)
+		$(CXX) $(TESTOBJS) $(UTILS)/cpuinfo.o $(MARCHFLAG) $(CXXFLAGS) -lgtest_main $(GTESTLDFLAGS) -o testexe
 
 bench: $(BENCHDIR)/main.cpp $(SRCS) $(UTILS)/cpuinfo.o
 		$(CXX) $(BENCHDIR)/main.cpp $(CXXFLAGS) $(UTILS)/cpuinfo.o $(MARCHFLAG) -o benchexe
 
+meson:
+	meson setup builddir && cd builddir && ninja
+
 clean:
-		rm -f $(TESTDIR)/*.o testexe benchexe
+	rm -rf $(TESTDIR)/*.o testexe benchexe builddir
```

Comment:
`$(RM) -r` instead of manually invoking `rm` is the common way of doing this.

Reply:
Makes sense.